odoo/addons/base/models/res_company.py

```diff
@@ -229,9 +229,24 @@ def create(self, vals_list):
 
         return companies
 
+    def cache_invalidation_fields(self):
+        # This list is not well defined and tests should be improved
+        return {
+            'active', # user._get_company_ids and other potential cached search
+            'sequence', # user._get_company_ids and other potential cached search
+        }
+
     def write(self, values):
-        self.clear_caches()
-        # Make sure that the selected currency is enabled
+        invalidation_fields = self.cache_invalidation_fields()
+        asset_invalidation_fields = {'font', 'primary_color', 'secondary_color', 'external_report_layout_id'}
+        if not invalidation_fields.isdisjoint(values):
+            self.env.registry.clear_cache()
+
+        if not asset_invalidation_fields.isdisjoint(values):
+            # this is used in the content of an asset (see asset_styles_company_report)
+            # and thus needs to invalidate the assets cache when this is changed
+            self.env.registry.clear_cache('assets')  # not 100% it is useful a test is missing if it is the case
+
```

Comment:
The problem with the previous invalidation was that it was quite difficult to identify what change on company will trigger an invalidation, and which part of the cache. 

Some causes where identified but it is quite hard to be sure that all cases where identified.

The main reason we want to avoid that is that a cron is writing on `hr_presence_last_compute_date` quite often invalidating the cache periodically for no reason.

Reply:
maybe in this case we could have a list of field that doesn't invalidate instead to guess the opposite.
Write on company is a rare case in theory (or bug like depends non readonly :fearful: ) and it will change a lot of cached record.

Reply:
> maybe in this case we could have a list of field that doesn't invalidate instead to guess the opposite. Write on company is a rare case in theory (or bug like depends non readonly fearful ) and it will change a lot of cached record.

This was discussed with @beledouxdenis and we choose this solution to have a fresh start.